common/datavalues/src/columns/column.rs

```diff
@@ -150,6 +151,9 @@ pub trait Column: Send + Sync {
 pub trait IntoColumn {
     fn into_column(self) -> ColumnRef;
     fn into_nullable_column(self) -> ColumnRef;
+
+    fn into_column_with_field(self, f: &ArrowField) -> ColumnRef;
```

Comment:
So `into_column` & `into_nullable_column` can be removed because they are covered by new methods?

Reply:
`into_column` & `into_nullable_column` need to be kept, because in some cases we don't have a field